### PR TITLE
Bayesian AB Test in jaspResults

### PR DESCRIPTION
**Pull request by akashrajkn**
1. Analysis in jaspResults
2. Informative footnote: Fixes jasp-stats/INTERNAL-jasp#505, jasp-stats/INTERNAL-jasp#367
3. Bayes factors Robustness plot: Fixes jasp-stats/INTERNAL-jasp#505

ab

### CODE REVIEW

**Comment by EJWagenmakers** [comment]
Looks awesome
Cheers,
E.J.
********************************************
Eric-Jan Wagenmakers
Department of Psychological Methods, room G 0.29
University of Amsterdam, Nieuwe Achtergracht 129B
Letter: PO Box 15906, 1001 NK Amsterdam
Parcel: Valckenierstraat 59,  1018 XE Amsterdam

Web: ejwagenmakers.com
Book: bayesmodels.com
Stats: jasp-stats.org
Blog: bayesianspectacles.org
Email: EJ.Wagenmakers@gmail.com
Cell: (+31) 6 45626624

“Man follows only phantoms.”
Pierre-Simon Laplace, last words
********************************************

**Comment by TimKDJ** [comment]
Can you rebase? The plot editing merge broke the unit tests, but they're fixed now

File: JASP-Engine/JASP/R/abtestbayesian.R [hunk 1]
```diff
@@ -16,283 +16,191 @@
...
-    if (status$error) {
-        table[["error"]] <- list(errorType = "badData", errorMessage = status$errorMessage)
-    }
+  if (!ready)
+    return()
+
+  .abTestBayesianFillTableMain(abTestBayesianTable, ab_obj, options)
 
-    return(table)
+  abTestBayesianTable$addFootnote("A positive log odds ratio means that the success rate in Group 2 is higher than in Group 1.")
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/abtestbayesian.R:113]: I don't understand this footnote. What column are you referring to?

File: JASP-Engine/JASP/R/abtestbayesian.R [hunk 2]
```diff
@@ -16,283 +16,191 @@
...
+      output.rows[[r]]$BF <- base::log(output.rows[[r]]$BF)
+    }
+  }
+
+  abTestBayesianTable$addRows(output.rows)
 }
 
 
-.calcModel.abTest <- function(dataset, status, options) {
+.calcModel.abTest <- function(dataset, options) {
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/abtestbayesian.R:199]: You should store this in a jaspState; right now it's being computed every time and it's making the analysis slower than it needs to be

**Review by TimKDJ — CHANGES_REQUESTED** [review]
Short and sweet; looks good!

Not sure what's up with the sequential plot though (new is at the bottom):
Screenshot 2019-10-31 at 13 20 20

File: JASP-Engine/JASP/R/abtestbayesian.R (hunk 1 above)
Comment (akashrajkn) [reply JASP-Engine/JASP/R/abtestbayesian.R:113]: This is a just an informative note for the AB Test table (https://github.com/jasp-stats/INTERNAL-jasp/issues/367)

File: JASP-Engine/JASP/R/abtestbayesian.R (hunk 2 above)
Comment (TimKDJ) [reply JASP-Engine/JASP/R/abtestbayesian.R:199]: @akashrajkn do you have time to incorporate this? Or shall I do it and merge this now

Comment (TimKDJ) [reply JASP-Engine/JASP/R/abtestbayesian.R:199]: @akashrajkn I'll merge this tomorrow and implement this myself unless you don't want me to

**Comment by TimKDJ** [comment]
@akashrajkn I will implement the state system. Did you and @quentingronau look at the plot?

**Pull request merged**